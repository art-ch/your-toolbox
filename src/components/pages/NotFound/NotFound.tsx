'use client';

import { Hero } from '@/components/Hero/Hero';
import { useTranslation } from 'react-i18next';

export const NotFound = () => {
  const { t } = useTranslation();

  return (
    <Hero
      title="404"
      subtitle={t('404:subtitle')}
      buttonText={t('common:checkAvailableTools')}
    />
  );
};
