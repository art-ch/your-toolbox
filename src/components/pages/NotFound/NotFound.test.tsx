import { render } from '@testing-library/react';

import { NotFound } from './NotFound';

describe('NotFound component', () => {
  it('should render correctly', () => {
    const { container } = render(<NotFound />);

    expect(container).toMatchSnapshot();
  });
});
