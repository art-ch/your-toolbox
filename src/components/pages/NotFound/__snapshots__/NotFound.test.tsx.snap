// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly 1`] = `
<div>
  <div
    class="my-0 mx-auto py-40 text-center max-w-lg"
  >
    <div
      class="my-0 mx-auto py-2"
    >
      <h1
        class="text-4xl sm:text-6xl font-semibold leading-none tracking-tight"
      >
        404
      </h1>
      <p
        class="py-2"
      >
        Your curiosity is astounding. Though, at the moment we don't have the treasure you seek
      </p>
    </div>
    <a
      href="/navigation"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 cursor-pointer"
        data-slot="button"
      >
        Check available tools
      </button>
    </a>
  </div>
</div>
`;
